### PR TITLE
refactor: Centralized annotations and labels, and updated doc

### PR DESCRIPTION
**Pull request by YanniHu1996**
closes: #2794

### CODE REVIEW

File: controllers/scheduledbackup_controller.go [hunk 1]
```diff
@@ -33,20 +33,11 @@ import (
...
-	"github.com/cloudnative-pg/cloudnative-pg/pkg/specs"
 	"github.com/cloudnative-pg/cloudnative-pg/pkg/utils"
 )
 
 const (
 	backupOwnerKey = ".metadata.controller"
-
-	// ImmediateBackupLabelName label is applied to backups to tell if a backup
-	// is immediate or not
-	ImmediateBackupLabelName = specs.MetadataNamespace + "/immediateBackup"
```
Comment (litaocdl) [inline controllers/scheduledbackup_controller.go:45]: not used by external

File: controllers/scheduledbackup_controller.go [hunk 2]
```diff
@@ -33,20 +33,11 @@ import (
...
-
-	// ParentScheduledBackupLabelName label is applied to backups to easily tell the scheduled backup
-	// it was created from.
-	ParentScheduledBackupLabelName = specs.MetadataNamespace + "/scheduled-backup"
```
Comment (litaocdl) [inline controllers/scheduledbackup_controller.go:49]: This is referenced by dppgo, adding it back

File: pkg/reconciler/persistentvolumeclaim/status.go [hunk 3]
```diff
@@ -35,10 +35,6 @@ import (
 type PVCStatus = string
 
 const (
-	// StatusAnnotationName is an annotation that shows the current status of the PVC.
-	// The status can be "initializing", "ready" or "detached"
-	StatusAnnotationName = specs.MetadataNamespace + "/pvcStatus"
```
Comment (litaocdl) [inline pkg/reconciler/persistentvolumeclaim/status.go:40]: not referenced outside in EnterpriseDB

File: pkg/specs/pgbouncer/deployments.go [hunk 4]
```diff
@@ -31,17 +31,11 @@ import (
...
 	"github.com/cloudnative-pg/cloudnative-pg/pkg/postgres"
 	"github.com/cloudnative-pg/cloudnative-pg/pkg/specs"
+	"github.com/cloudnative-pg/cloudnative-pg/pkg/utils"
 	"github.com/cloudnative-pg/cloudnative-pg/pkg/utils/hash"
 )
 
 const (
-	// PgbouncerPoolerSpecHash is the annotation added to the deployment to tell
-	// the hash of the Pooler Specification
-	PgbouncerPoolerSpecHash = specs.MetadataNamespace + "/poolerSpecHash"
```
Comment (litaocdl) [inline pkg/specs/pgbouncer/deployments.go:40]: not used external in enterprisedb org

File: pkg/specs/pgbouncer/deployments.go [hunk 5]
```diff
@@ -31,17 +31,11 @@ import (
...
-
-	// PgbouncerNameLabel is the label of the pgbouncer pod used by default
-	PgbouncerNameLabel = specs.MetadataNamespace + "/poolerName"
```
Comment (litaocdl) [inline pkg/specs/pgbouncer/deployments.go:43]: not used external in enterprisedb org

File: pkg/specs/pods.go [hunk 6]
```diff
@@ -38,33 +38,12 @@ import (
 )
 
 const (
-	// MetadataNamespace is the annotation and label namespace used by the operator
```
Comment (litaocdl) [inline pkg/specs/pods.go:41]: Going to add those public constants back as it is referenced.

File: pkg/specs/serviceaccount.go [hunk 7]
```diff
@@ -25,12 +25,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/cloudnative-pg/cloudnative-pg/pkg/management/log"
-)
-
-const (
-	// OperatorManagedSecretsName is the name of the annotation containing the secrets
-	// managed by the operator inside the generated service account
-	OperatorManagedSecretsName = "cnpg.io/managedSecrets" // #nosec
```
Comment (litaocdl) [inline pkg/specs/serviceaccount.go:33]: not used external

File: pkg/utils/fencing.go [hunk 8]
```diff
@@ -45,11 +45,6 @@ var (
 )
 
 const (
-	// FencedInstanceAnnotation is the annotation to be used for fencing instances, the value should be a
-	// JSON list of all the instances we want to be fenced, e.g. `["cluster-example-1","cluster-example-2`"].
-	// If the list contain the "*" element, every node is fenced.
-	FencedInstanceAnnotation = "cnpg.io/fencedInstances"
```
Comment (litaocdl) [inline pkg/utils/fencing.go:51]: not used external

File: tests/e2e/pgbouncer_types_test.go [hunk 9]
```diff
@@ -22,7 +22,7 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
 
-	"github.com/cloudnative-pg/cloudnative-pg/pkg/specs/pgbouncer"
+	utils2 "github.com/cloudnative-pg/cloudnative-pg/pkg/utils"
```
Comment (phisco) [inline tests/e2e/pgbouncer_types_test.go:25]: definitely a non blocker nit: alias this one to pkgutils too.

Comment (litaocdl) [reply tests/e2e/pgbouncer_types_test.go:25]: Thank you!

File: pkg/utils/labels_annotations.go [hunk 10]
```diff
@@ -24,65 +24,131 @@ import (
...
+	// PostgreSQL cluster
+	HibernationAnnotationName = MetadataNamespace + "/hibernation"
+
+	// PoolerSpecHashAnnotationName is the name of the annotation added to the deployment to tell
+	// the hash of the Pooler Specification
+	PoolerSpecHashAnnotationName = MetadataNamespace + "/poolerSpecHash"
+
+	// OperatorManagedSecretsAnnotationName is the name of the annotation containing
+	// the secrets managed by the operator inside the generated service account
+	OperatorManagedSecretsAnnotationName = MetadataNamespace + "/managedSecrets" // #nosec
```
[the remark this answers is not in the document]
Comment (litaocdl) [reply pkg/utils/labels_annotations.go:138]: +1

File: controllers/scheduledbackup_controller.go (hunk 2 above)
Comment (litaocdl) [reply controllers/scheduledbackup_controller.go:49]: already removed

**Comment by litaocdl** [comment]
@phisco @gbartolini The comments is addressed and pr rebased

**Review by litaocdl — APPROVED** [review]

**Review by gbartolini — APPROVED** [review]

**Pull request merged**